exporter/exporterhelper/traces_batch.go

```diff
@@ -7,13 +7,15 @@ import (
 	"context"
 	"errors"
 
+	"go.uber.org/zap"
+
 	"go.opentelemetry.io/collector/exporter/exporterhelper/internal/sizer"
 	"go.opentelemetry.io/collector/pdata/ptrace"
 )
 
 // MergeSplit splits and/or merges the provided traces request and the current request into one or more requests
 // conforming with the MaxSizeConfig.
-func (req *tracesRequest) MergeSplit(_ context.Context, maxSize int, szt RequestSizerType, r2 Request) ([]Request, error) {
+func (req *tracesRequest) MergeSplit(_ context.Context, maxSize int, szt RequestSizerType, r2 Request, logger *zap.Logger) ([]Request, error) {
```

Comment:
Adding the argument breaks the public API. Why can't we return the error and log it in the `Consume` method?

If we don't want to drop the whole batch, we can introduce new type of errors that would represent the OTLP PartialSuccess.

Reply:
Mmmm good idea! Let me take a look.
Thanks!